### PR TITLE
Documentation for audio classes

### PR DESCRIPTION
**Pull request by f-brinkmann**
I suggest that you check the documentation in the html version after building it (see CONTRIBUTIONS.rst). Feel free to directly commit fixes for typos and minor issues.

I included inherited methods in the documentation and added documentation for magic methods `__iter__`, `__getitem__`, and `__setitem__`.

I touched one thing not related to the documentation
- made SignalIterator a private class

### CODE REVIEW

File: pyfar/signal.py [hunk 1]
```diff
@@ -376,15 +401,16 @@ def __init__(self, data, frequencies, fft_norm=None, comment=None,
...
-            pyfar.fft.normalization and _[#] for more information. The default
-            is 'none', which is typically used for energy signals, such as
-            impulse responses.
-        comment : str
+        fft_norm : str, optional
+            The normalization of the Discrete Fourier Transform (DFT). Can be
+            'none', 'unitary', 'amplitude', 'rms', 'power', or 'psd'. See
+            :py:func:`pyfar.dsp.fft.normalization` and [#]_ for more
+            information. The default is 'none', which is typically used for
+            energy signals, such as impulse responses.
```
Comment (sikersten) [inline pyfar/signal.py:409]: Comment on why `fft_norm` it is useful for FrequencyData would be nice

File: pyfar/signal.py [hunk 2]
```diff
@@ -769,13 +802,26 @@ def __len__(self):
         return self.n_samples
 
     def __iter__(self):
-        """Iterator for signals. The actual iteration is handled through
-        numpy's array iteration.
+        """Iterator for audio objects.
+
+        Iterate across the first dimension of an audio object. The actual
+        iteration is handled through numpy's array iteration.
```
Comment (sikersten) [inline pyfar/signal.py:807]: Only used by Signal, adjust docstring?

File: pyfar/signal.py [hunk 3]
```diff
@@ -124,9 +129,9 @@ def reshape(self, newshape):
         reshaped : Signal
```
Comment (sikersten) [inline pyfar/signal.py]: Could also be FrequencyData or TimeData, same holds for flatten()

File: pyfar/signal.py [hunk 4]
```diff
@@ -800,135 +846,141 @@ def __next__(self):
...
-        Data to be added. Can contain Signal objects, numbers, and, array
-        likes.
-    domain : 'time', 'freq'
+        Data to be added. Can contain pyfar audio objects, array likes, and
+        scalars. Pyfar audio objects can not be mixed, e.g., TimeData and
+        FrequencyData objects do not work together.
+    domain : 'time', 'freq', optional
         Flag to indicate if the operation should be performed in the time or
         frequency domain. If working in the frequency domain, the FFT
         normalization is removed before the operation (See
```
Comment (sikersten) [inline pyfar/signal.py:861]: removed = set to 'none'? Is this information important for the user? same holds for other arithmetic operations

**Review by sikersten — COMMENTED** [review]
- Commited some minor changes, mainly replaced "signal" with "audio object" at some positions
- Formatting of references is not consistent with e.g. pyfar.signals. Let's agree on a format for all references?
- I don't quite get when to use ``attribute`` with `` and `attribute` with `, what's the rule?
- self is used in docstrings, see comments in review on #139

File: pyfar/signal.py (hunk 4 above)
Comment (f-brinkmann) [reply pyfar/signal.py:861]: No, I also it is better this way

File: pyfar/signal.py (hunk 2 above)
Comment (f-brinkmann) [reply pyfar/signal.py:807]: yes, did that.

**Pull request merged**